### PR TITLE
Add STM32 gdb server configuration

### PR DESCRIPTION
**Pull request by Kneicik**
Add variable RUN_STM32 for debugging and flashing STM32 MCUs

Currently img size needs to be 16G in order for root filesystem not to run out of free space. 
Need to resize root filesystem and shrink the partition in the future

### CODE REVIEW

**Review by pgielmuda — CHANGES_REQUESTED** [review]
Please resolve conflicts by **rebasing** branch onto the main branch!

Beside that I constantly get error as follow:
`After this operation, 61.5 MB of additional disk space will be used.
Ign:1 http://deb.debian.org/debian bookworm/main arm64 exim4-config all 4.96-15+deb12u4
Ign:2 http://deb.debian.org/debian bookworm/main arm64 exim4-base arm64 4.96-15+deb12u4
Ign:1 http://deb.debian.org/debian bookworm/main arm64 exim4-config all 4.96-15+deb12u4
Err:2 http://deb.debian.org/debian bookworm/main arm64 exim4-base arm64 4.96-15+deb12u4
  404  Not Found [IP: 199.232.18.132 80]
Ign:1 http://deb.debian.org/debian bookworm/main arm64 exim4-config all 4.96-15+deb12u4
Err:1 http://deb.debian.org/debian bookworm/main arm64 exim4-config all 4.96-15+deb12u4
  404  Not Found [IP: 199.232.18.132 80]
E: Failed to fetch http://deb.debian.org/debian-security/pool/updates/main/e/exim4/exim4-config_4.96-15%2bdeb12u4_all.deb  404  Not Found [IP: 199.232.18.132 80]
E: Failed to fetch http://deb.debian.org/debian-security/pool/updates/main/e/exim4/exim4-base_4.96-15%2bdeb12u4_arm64.deb  404  Not Found [IP: 199.232.18.132 80]
E: Unable to fetch some archives, maybe run apt-get update or try with --fix-missing?
+ echo 'Installing dependencies failed'`

Any ideas?

File: remoterover.sh [hunk 1]
```diff
@@ -280,13 +289,19 @@ if [[ "x${RUN_SERIAL}" == "x1" ]]; then
   ) || die "Installing Minicom failed"
 fi
 
-if [[ "x${RUN_RASPICONF}" == "x1" ]]; then
-  info "Running raspi-config commands"
```
Comment (pgielmuda) [inline remoterover.sh:284]: Can you explain what is the purpose of this change?

File: remoterover.sh [hunk 2]
```diff
@@ -280,13 +289,19 @@ if [[ "x${RUN_SERIAL}" == "x1" ]]; then
...
+if [[ "x${RUN_STM32}" == "x1" ]]; then
+  info "Enabling STM32"
   ( set -x
-  for cmd in "${RASPICMDS[@]}"; do
```
Comment (pgielmuda) [inline remoterover.sh:286]: Same here - why do we delete that?

Comment (Kneicik) [reply remoterover.sh:286]: RASPICONF was moved up because expanding filesystem needs to be done first after booting the system. Main branch didn't have this change so it made this weird code chimera. If you look at line 250 everything is in place
